src/ios/CDVLocation.m

```diff
@@ -169,199 +180,221 @@ - (void)_stopLocation
+
+    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsDictionary:locationInfo];
+    [result setKeepCallbackAsBool:keepCallback];
+
+    [self.commandDelegate sendPluginResult:result callbackId:callbackId];
 }
 
-- (void)returnLocationError:(NSUInteger)errorCode withMessage:(NSString*)message
+- (void)returnLocationError:(NSUInteger)errorCode withMessage:(NSString *)message
 {
-    NSMutableDictionary* posError = [NSMutableDictionary dictionaryWithCapacity:2];
+    NSMutableDictionary *errorInfo = [NSMutableDictionary dictionaryWithCapacity:2];
+    errorInfo[@"code"] = @(errorCode);
+    errorInfo[@"message"] = message ?: @"";
 
-    [posError setObject:[NSNumber numberWithUnsignedInteger:errorCode] forKey:@"code"];
-    [posError setObject:message ? message:@"" forKey:@"message"];
-    CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:posError];
-
-    for (NSString* callbackId in self.locationData.locationCallbacks) {
-        [self.commandDelegate sendPluginResult:result callbackId:callbackId];
-    }
+    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:errorInfo];
 
+    [self sendResultToCallbacks:result callbacks:self.locationData.locationCallbacks];
     [self.locationData.locationCallbacks removeAllObjects];
 
-    for (NSString* callbackId in self.locationData.watchCallbacks) {
+    [self sendResultToCallbacks:result callbacks:self.locationData.watchCallbacks.allValues];
+}
+
+- (void)sendResultToCallbacks:(CDVPluginResult *)result callbacks:(NSArray<NSString *> *)callbacks
+{
+    for (NSString *callbackId in callbacks) {
         [self.commandDelegate sendPluginResult:result callbackId:callbackId];
     }
 }
 
-- (void)locationManager:(CLLocationManager*)manager didFailWithError:(NSError*)error
+- (void)sendPluginResultError:(NSUInteger)errorCode callbackId:(NSString *)callbackId
 {
-    NSLog(@"locationManager::didFailWithError %@", [error localizedFailureReason]);
-
-    CDVLocationData* lData = self.locationData;
-    if (lData && __locationStarted) {
-        // TODO: probably have to once over the various error codes and return one of:
-        // PositionError.PERMISSION_DENIED = 1;
-        // PositionError.POSITION_UNAVAILABLE = 2;
-        // PositionError.TIMEOUT = 3;
-        NSUInteger positionError = POSITIONUNAVAILABLE;
-        if (error.code == kCLErrorDenied) {
-            positionError = PERMISSIONDENIED;
-        }
-        [self returnLocationError:positionError withMessage:[error localizedDescription]];
+    NSMutableDictionary *errorInfo = [NSMutableDictionary dictionaryWithCapacity:2];
+    errorInfo[@"code"] = @(errorCode);
+    errorInfo[@"message"] = @"";
+
+    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:errorInfo];
+    [self.commandDelegate sendPluginResult:result callbackId:callbackId];
+}
```

Comment:
This code is almost a duplicate of `returnLocationError:withMessage:`. Can't we just reuse one of them as needed?

Reply:
I can extract the code in common 

`NSMutableDictionary *errorInfo = [NSMutableDictionary dictionaryWithCapacity:2];
    errorInfo[@"code"] = @(errorCode);
    errorInfo[@"message"] = message ?: @"";`

but I would keep the two methods

Reply:
As long as we avoid duplication, the way to do it is irrelevant.

Reply:
Fixed.